scripts/dependency-graph-generator/index.js

```diff
@@ -0,0 +1,174 @@
+const spawnSync = require('child_process').spawnSync;
+const findGitRoot = require('../monorepo/index').findGitRoot;
+const getDevDependencies = require('./getDevDependencies');
+const path = require('path');
+const os = require('os');
+
+const dotFilePath = path.resolve(__dirname, 'repo-graph.dot');
+let ignoreDevDependencies = [];
+
+/**
+ * Pick a package in the repository and generate its dependency graph
+ *
+ * For this utility to work you will actually need to install graphviz on your machine
+ * http://www.graphviz.org
+ */
+async function main(argv) {
+  if (!argv.root) {
+    throw new Error('A root package name should be provided');
+  }
+
+  const rootPackage = `@fluentui/${argv.root}`;
+
+  // if dev dependencies should be added
+  const includeDevDependencies = argv['include-dev'];
+  if (!includeDevDependencies) {
+    ignoreDevDependencies = await getDevDependencies(rootPackage);
+  }
+
+  _generateGraphforRepo(dotFilePath);
+  const graph = _parseDotFile(dotFilePath);
+  const subTree = _getSubTree(graph, rootPackage);
+
+  if (process.platform === 'win32') {
+    throw new Error('--graphvis-path argument is required for windows users');
+  }
+
+  subTree.setGraphVizPath(argv['graphviz-path'] || '/usr/bin');
+  const pngOuputFile = path.resolve(__dirname, `${argv.root}.png`);
+
+  subTree.output('png', pngOuputFile);
+}
+
+/**
+ * Generates a dependency graph of the entire repo as a dotfile
+ * For this function to work you will actually need to install graphviz on your machine
+ * http://www.graphviz.org
+ *
+ * @param {string} outputPath the output path of the dot file
+ */
+function _generateGraphforRepo(outputPath) {
+  const lernaDependencyGraphArgs = [`--outputPath ${outputPath}`];
+  // The package uses a shebang with windows line endings, call node on the entrypoint directly to avoid issues
+  const lernaDependencyGraphEntry = path.resolve(findGitRoot(), require.resolve('lerna-dependency-graph/lib/index.js'));
+
+  const result = spawnSync(`node ${lernaDependencyGraphEntry}`, lernaDependencyGraphArgs, {
+    cwd: findGitRoot(),
+    shell: true,
+    stdio: 'inherit',
+  });
+
+  if (result.status) {
+    throw new Error(
+      (result.error && result.error.stack) || `failed to run lerna-dependency-graph with status ${result.status}`,
+    );
+  }
+}
+
+/**
+ * Returns an graphviz instance of a parsed dotfile
+ * Assumes one graph only in the dotfile
+ *
+ * @param {string} pathToDotFile
+ */
+function _parseDotFile(pathToDotFile) {
+  const content = readFileSync(pathToDotFile);
+  const graph = graphviz.digraph('G');
+  const parsed = parser(content.toString());
+
+  const items = parsed[0].children;
+  items.forEach(item => {
+    if (item.type === 'node_stmt') {
+      graph.addNode(item.node_id.id);
+    } else {
+      graph.addEdge(item.edge_list[0].id, item.edge_list[1].id);
+    }
+  });
+
+  return graph;
+}
+
+/**
+ *
+ * @param {graphviz graph} graph The original graph
```

Comment:
For the type here you might be able to do `import("graphviz").Whatever`. Same for the next parameter if it's some node type not a string.

Reply:
good call